api/v1/agentconfig_types_test.go

```diff
@@ -370,6 +370,28 @@ func TestAgentConfigSpecAdapter_GetRetryLimit(t *testing.T) {
 	}
 }
 
+func TestAgentConfigSpecAdapter_GetTTLSecondsAfterFinished(t *testing.T) {
+	// var testdataDefault int32 = 600
+	var testdatauserSubmitted int32 = 700
+	testCases := []struct {
+		name                    string
+		TTLSecondsAfterFinished *int32
+		expected                *int32
+	}{
+		// {name: "default", TTLSecondsAfterFinished: nil, expected: &testdataDefault},
+		{name: "user submitted", TTLSecondsAfterFinished: &testdatauserSubmitted, expected: &testdatauserSubmitted},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			adapter := NewAgentConfigSpecAdapter(AgentConfigSpec{
+				TTLSecondsAfterFinished: tc.TTLSecondsAfterFinished,
+			})
+			result := adapter.GetTTLSecondsAfterFinished()
```

Comment:
This is testing that your getter and your setter work.  

I'm not sure but if you test the creation of this resource without setting it does it show that this is set with the default? _(Probably not unit wise but with maybe env test, because env test uses a fake api-server and loads the crds)_

What worries me is this:
You have an `agentConfig`, you have an `agentConfigAdapter`, you have an `agentConfigSpecAdapter`. 

If you are putting this on the agentConfig, those things will get set by the nature of them copying the existing structure over to the new "type".  Satisfying the other types, _(yes you need the Get method to get the TTL set on the original agentconfig)_ makes this more complicated that it has to be.  I would imagine that the api/v1 type (the agentconfig) is the one with the owner reference and needs to have that information on it to pass to the batch/v1 job to settle the timeout. The adapter extending this doesn't seem consistent with what  putting the type on the agentConfig. 

TLDR; to not change anything with the crds, you can just use the non exposed api/v1 surface _(agenconfigSpecAdapter)_ and you can perform this same functionality without changing the schema for the api that exists.  That way you don't have to do anything with agentconfig besides add the owner on it.

Reply:
So it may be better served to not add it to the agentconfig type but to the agentconfigspecAdapter type.